wpiutil/src/main/native/cpp/MimeTypes.cpp

```diff
@@ -0,0 +1,69 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "wpi/MimeTypes.h"
+
+#include "wpi/StringMap.h"
+
+using namespace wpi;
+
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types
+static StringMap<std::string> mimeTypes{
```

Comment:
`StringMap<const char*>`.  Move this entire thing into the body of MimeTypeFromPath to avoid adding to static initializers (this will delay initialization until the first call to MimeTypeFromPath, so the initialization penalty is only paid if the function is actually called).